lib/jekyll/renderer.rb

```diff
@@ -196,10 +196,7 @@ def render_layout(output, layout, info)
     private
     def add_regenerator_dependencies(layout)
       return unless document.write?
-      site.regenerator.add_dependency(
-        site.in_source_dir(document.path),
-        site.in_source_dir(layout.path)
-      )
+      site.regenerator.add_dependency(document.path, layout.path)
```

Comment:
I'm worried that this change could introduce security issues because we're not being explicit about files being in the site's source directory. Do we have tests that indicate that document and layout objects can only exist within the sandbox that is a jekyll site?

Reply:
I agree with your concerns regarding security since this change removes the existing sanitization..
While `document` here refers to any renderable object (`Jekyll::Document`, `Jekyll::Excerpt`, `Jekyll::Page`, `Jekyll::PageWithoutAFile`) which can be manually created by a plugin and pushed into the site payload via the `:pre_render` hooks, this method here **simply considers adding the concerned paths** to the metadata file when required..

Layouts can exist both in the `site.source` and in `site.theme.root` elsewhere. So coercing a `layout.path` to be strictly within the `site.source` just here is not right.
https://github.com/jekyll/jekyll/blob/999151dcaad7209254a4513f256ef87612690eff/lib/jekyll/layout.rb#L38-L44

I can coerce the `document.path` to be within the `site.source` and not affect this patch at all since the `document.path` is simply a `key` in the `metadata` hash..